frontend/amundsen_application/static/js/components/Table/index.tsx

```diff
@@ -230,15 +237,41 @@ const getKeysToExpand = (preExpandPanelKey, tableKey) => {
 
 const getInitialExpandedRows = (
   data,
+  allColumnKeys,
   maxNumRows,
   preExpandPanelKey,
   tableKey
-) => {
-  const allColumnKeys = getAllColumnKeys(data);
-
-  return allColumnKeys.length <= maxNumRows
+) =>
+  allColumnKeys.length <= maxNumRows
     ? allColumnKeys
     : getKeysToExpand(preExpandPanelKey, tableKey);
+
+const expandOrCollapseAllRows = (
+  data,
+  allColumnKeys,
+  shouldExpandAllRows,
+  setExpandedRows,
+  initialExpandedRows,
+  maxNumRows,
+  toggleExpandingRows
```

Comment:
Let's use an object for input parameters once they get this long.
The benefit is that we won't need to remember the order of them.